### DISCUSSION TITLE
Redirecting to arbitrary route in auth0 callback

### DISCUSSION THREAD

**Discussion by aloukissas** [vercel/next.js#11555] [Help]
Hi folks,

I'm building a small app, based on the auth0 sample. What I haven't found out how to do is how to redirect to an arbitrary route in the post-login callback. For example, assume there is a protected route `/account` that I redirect to `/api/login` as in the sample. I would like to pass a url parameter (e.g. `?redirect=/account`) that the `/api/callback` route takes and redirects me to `/account`. Is that possible? It's a pretty common pattern in many web apps.

**Comment by aloukissas** [comment]
So it looks like the auth0 library supports this (at least their docs say they do), but it doesn't seem to work in practice (i.e. you can pass a `redirectTo` parameter in the login handler, but this is largely ignored).

**Comment by waptik** [comment]
This might be what you want https://www.github.com/ThatConference/that-website/tree/master/pages%2Fapi%2Fcallback.js

**Reply by aloukissas** [reply]
This is promising, thanks!
